x/gamm/types/pool.go

```diff
@@ -3,60 +3,16 @@ package types
 import (
 	"time"
 
-	proto "github.com/gogo/protobuf/proto"
-
 	"github.com/cosmos/cosmos-sdk/types/address"
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
-)
 
-// PoolI defines an interface for pools that hold tokens.
-type PoolI interface {
-	proto.Message
-
-	GetAddress() sdk.AccAddress
-	String() string
-	GetId() uint64
-	// GetSwapFee returns the pool's swap fee, based on the current state.
-	// Pools may choose to make their swap fees dependent upon state
-	// (prior TWAPs, network downtime, other pool states, etc.)
-	// hence Context is provided as an argument.
-	GetSwapFee(ctx sdk.Context) sdk.Dec
-	// GetExitFee returns the pool's exit fee, based on the current state.
-	// Pools may choose to make their exit fees dependent upon state.
-	GetExitFee(ctx sdk.Context) sdk.Dec
-	// Returns whether the pool has swaps enabled at the moment
-	IsActive(ctx sdk.Context) bool
-	// GetTotalShares returns the total number of LP shares in the pool
-	GetTotalShares() sdk.Int
-
-	// SwapOutAmtGivenIn swaps 'tokenIn' against the pool, for tokenOutDenom, with the provided swapFee charged.
-	// Balance transfers are done in the keeper, but this method updates the internal pool state.
-	SwapOutAmtGivenIn(ctx sdk.Context, tokenIn sdk.Coins, tokenOutDenom string, swapFee sdk.Dec) (tokenOut sdk.Coin, err error)
-	// CalcOutAmtGivenIn returns how many coins SwapOutAmtGivenIn would return on these arguments.
-	// This does not mutate the pool, or state.
-	CalcOutAmtGivenIn(ctx sdk.Context, tokenIn sdk.Coins, tokenOutDenom string, swapFee sdk.Dec) (tokenOut sdk.Coin, err error)
-
-	// SwapInAmtGivenOut swaps exactly enough tokensIn against the pool, to get the provided tokenOut amount out of the pool.
-	// Balance transfers are done in the keeper, but this method updates the internal pool state.
-	SwapInAmtGivenOut(ctx sdk.Context, tokenOut sdk.Coins, tokenInDenom string, swapFee sdk.Dec) (tokenIn sdk.Coin, err error)
-	// CalcInAmtGivenOut returns how many coins SwapInAmtGivenOut would return on these arguments.
-	// This does not mutate the pool, or state.
-	CalcInAmtGivenOut(ctx sdk.Context, tokenOut sdk.Coins, tokenInDenom string, swapFee sdk.Dec) (tokenIn sdk.Coin, err error)
-
-	// Returns the spot price of the 'base asset' in terms of the 'quote asset' in the pool,
-	// errors if either baseAssetDenom, or quoteAssetDenom does not exist.
-	// For example, if this was a UniV2 50-50 pool, with 2 ETH, and 8000 UST
-	// pool.SpotPrice(ctx, "eth", "ust") = 4000.00
-	SpotPrice(ctx sdk.Context, baseAssetDenom string, quoteAssetDenom string) (sdk.Dec, error)
-}
+	swaproutertypes "github.com/osmosis-labs/osmosis/v13/x/swaprouter/types"
+)
 
 // TraditionalAmmInterface defines an interface for pools representing traditional AMM.
 type TraditionalAmmInterface interface {
-	PoolI
-
-	// GetTotalPoolLiquidity returns the coins in the pool owned by all LPs
-	GetTotalPoolLiquidity(ctx sdk.Context) sdk.Coins
```

Comment:
Note: this was moved to `PoolI`. Although this getter might not be very useful for CL pools, it is still possible to have it. In the context of this PR, moving it simplifies the refactor here if the method is defined on `PoolI`. As a result, I would like to revisit this later